provider/openrouter/openrouter.go

```diff
@@ -29,11 +29,17 @@ func newOpenRouterProvider() provider.Provider {
 	}
 }
 
-// Query queries the LLM.
-func (o *openRouterProvider) Query(ctx context.Context, modelIdentifier string, promptText string) (response string, err error) {
+// client returns a new client with the current configuration.
+func (o *openRouterProvider) client() (client *openai.Client) {
 	config := openai.DefaultConfig(o.token)
 	config.BaseURL = o.baseURL
-	client := openai.NewClientWithConfig(config)
+
+	return openai.NewClientWithConfig(config)
+}
+
+// Query queries the LLM.
+func (o *openRouterProvider) Query(ctx context.Context, modelIdentifier string, promptText string) (response string, err error) {
+	client := o.client()
```

Comment:
No need to instantiated the client even... we could do `o.client().CreateChatCompletion...` directly.

Reply:
Prefer a client object for a clean configuration.